parse.js

```diff
@@ -6,40 +6,33 @@
 	License: Apache-2.0
 */
 
-const ARGUMENT_SEPARATION_REGEX = /([^=\s]+)=?\s*(.*)/;
+const ARGUMENT_SEPARATION_REGEX = /^([^=\s]+)=?\s*(.*)/;
 
 function Parse(argv) {
-  // Removing node/bin and called script name
-  argv = argv.slice(2);
-
+  
   const parsedArgs = {};
-  let argName, argValue;
-
-  argv.forEach(function (arg) {
-    // Separate argument for a key/value return
-    arg = arg.match(ARGUMENT_SEPARATION_REGEX);
-    arg.splice(0, 1);
-
-    // Retrieve the argument name
-    argName = arg[0];
-
+  // Removing node/bin and called script name by starting at index 2
+  for(let i = 2; i < argv.length; i++){
+    // Separate argument for a key/value return, ignoring full match
+    let [_, argName, argValue] = argv[i].match(ARGUMENT_SEPARATION_REGEX);
+    
     // Remove "--" or "-"
-    if (argName.indexOf('-') === 0) {
-      argName = argName.slice(argName.slice(0, 2).lastIndexOf('-') + 1);
+    if (argName[0] === '-') {
+      argName = argName.slice(argName[1] === '-' ? 2 : 1);
     }
-
+    
     // Parse argument value or set it to `true` if empty
-    argValue =
-      arg[1] !== ''
-        ? parseFloat(arg[1]).toString() === arg[1]
-          ? +arg[1]
-          : arg[1]
+    const numberValue = +argValue
+    parsedArgs[argName] =
+      argValue !== ''
+        ? !Number.isNaN(numberValue) ? numberValue : argValue
         : true;
-
-    parsedArgs[argName] = argValue;
-  });
+  }
 
   return parsedArgs;
 }
 
 module.exports = Parse;
+
+// If process.argv is available, make parsed 
+if(typeof process !== 'undefined') module.exports.parsedArgv = Parse(process.argv || [])
```

Comment:
If you want to make sure `process` is defined, you'll need a try/catch instead

The reason is because if `process` is not globally defined, the script will crash

```js
module.exports.parsedArgv = function parsedArgv() {
  try {
   return Parse(process.argv || []);
  } catch  {
    console.error("parsedArgv is not available because process.argv is not defined in this environment.");
  }
};
```

Reply:
typeof does not throw an error if a variable does not exist, and simply returns "undefined" instead.